minisite.module

```diff
@@ -31,8 +20,35 @@ function minisite_help($route_name, RouteMatchInterface $route_match) {
     case 'help.page.minisite':
       $output = '';
       $output .= '<h3>' . t('About') . '</h3>';
-      $output .= '<p>' . t('Provides the ability to upload arbitrary \'minisites\' to a Drupal website.') . '</p>';
+      $output .= '<p>' . t('Upload and serve arbitrary "minisites" archives from a Drupal website.') . '</p>';
 
       return $output;
   }
 }
+
+/**
+ * Validation callback.
+ *
+ * @param \Drupal\file\FileInterface $file
+ *   The file to validate.
+ * @param string $content_extensions
+ *   String list of the allowed archive content extensions.
+ *
+ * @return array
+ *   Array of errors, if any.
+ */
+function minisite_validate_archive(FileInterface $file, $content_extensions) {
```

Comment:
upload validation callback used in widget when the archive is uploaded. Unfortunately, Drupal does not support OOP callbacks in `#upload_validation`, so this has to dangle here